tests/org.jboss.tools.fuse.ui.bot.test/src/org/jboss/tools/fuse/ui/bot/test/FeaturesServerTest.java

```diff
@@ -0,0 +1,105 @@
+
+import java.io.File;
+
+import org.jboss.reddeer.common.logging.Logger;
+import org.jboss.reddeer.common.wait.AbstractWait;
+import org.jboss.reddeer.common.wait.TimePeriod;
+import org.jboss.reddeer.common.wait.WaitUntil;
+import org.jboss.reddeer.common.wait.WaitWhile;
+import org.jboss.reddeer.core.condition.ShellWithTextIsAvailable;
+import org.jboss.reddeer.core.handler.ShellHandler;
+import org.jboss.reddeer.eclipse.wst.server.ui.view.ServersView;
+import org.jboss.reddeer.junit.runner.RedDeerSuite;
+import org.jboss.reddeer.requirements.cleanworkspace.CleanWorkspaceRequirement.CleanWorkspace;
+import org.jboss.reddeer.requirements.openperspective.OpenPerspectiveRequirement.OpenPerspective;
+import org.jboss.reddeer.swt.impl.button.PushButton;
+import org.jboss.reddeer.swt.impl.button.RadioButton;
+import org.jboss.reddeer.swt.impl.text.DefaultText;
+import org.jboss.tools.common.reddeer.FileUtils;
+import org.jboss.tools.fuse.reddeer.perspectives.FuseIntegrationPerspective;
+import org.jboss.tools.runtime.reddeer.preference.FuseServerRuntimePreferencePage;
+import org.jboss.tools.runtime.reddeer.preference.JBossRuntimeDetection;
+import org.jboss.tools.runtime.reddeer.wizard.DownloadRuntimesWizard;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Class contains test cases on a variety of server feature requests
+ * 
+ * @author djelinek
+ */
+@CleanWorkspace
+@OpenPerspective(FuseIntegrationPerspective.class)
+@RunWith(RedDeerSuite.class)
+public class FeaturesServerTest extends DefaultTest {
+
+	protected Logger log = Logger.getLogger(CamelEditorTest.class);
+
+	@Before
+	public void clean() {
+		// clean previous runtime installations
+		FileUtils.deleteDir(new File(getPath() + "target/temp_install/jboss-fuse-6.3.0.redhat-187"));
+		// close all shells
+		ShellHandler.getInstance().closeAllNonWorbenchShells();
+	}
+
+	/**
+	 * <p>
+	 * Fuse runtime detection never makes a server
+	 * <p>
+	 * <b>Link: </b>
+	 * <a href="https://issues.jboss.org/browse/FUSETOOLS-2176">https://issues.jboss.org/browse/FUSETOOLS-2176</a>
+	 */
+	@Test
+	public void issue_2176() {
+		downloadRuntime("JBoss Fuse 6.3.0");
+		assertTrue("Server was not created in view", new ServersView().getServers().size() == 1);
+		FuseServerRuntimePreferencePage runtimePref = new FuseServerRuntimePreferencePage();
+		runtimePref.open();
+		assertTrue("Server runtime was not created", runtimePref.getServerRuntimes().size() == 1);
+	}
+
+	/**
+	 * Downloads/installs new runtime into temp/temp_install folder
+	 * 
+	 * @param name
+	 *            String name of runtime
+	 */
+	private void downloadRuntime(String name) {
+		JBossRuntimeDetection prefPage = new JBossRuntimeDetection();
+		prefPage.open();
+		DownloadRuntimesWizard runtimeWiz = prefPage.downloadRuntime();
+		runtimeWiz.selectRuntime(name);
+		runtimeWiz.next();
+		new PushButton("Add...").click();
+		new DefaultText(0).setText("aqtwrdnr");
```

Comment:
Please use labels instead of indices. If this is not possible, please explain why.